core/tauri-utils/src/acl/capability.rs

```diff
@@ -129,6 +159,33 @@ impl CapabilityFile {
   }
 }
 
+#[derive(Deserialize, Debug)]
+struct CapabilityNamedList {
+  capabilities: Vec<Capability>,
+}
+
+impl<'de> Deserialize<'de> for CapabilityFile {
+  fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+  where
+    D: Deserializer<'de>,
+  {
+    UntaggedEnumVisitor::new()
+      .seq(|seq| seq.deserialize::<Vec<Capability>>().map(Self::List))
+      .map(|map| {
+        if let Some(len) = map.size_hint() {
+          if len == 1 {
```

Comment:
instead of checking the length, we could just try to deserialize as `CapabilityNamedList` and if it fails, we fallback to deserializing as `Capability`? I know this is less performant but should cover the case where `map.size_hint()` is None.

Reply:
looks like we can't because trying `.deserialize` takes ownership of the value and we can't retry again